x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/bulk_actions/bulk_edit_rules.ts

```diff
@@ -47,13 +48,17 @@ export const bulkEditRules = async ({
     ...(ids ? { ids } : { filter: enrichFilterWithRuleTypeMapping(filter) }),
     operations,
     paramsModifier: async (ruleParams: RuleAlertType['params']) => {
+      const migratedRuleParams = {
+        ...ruleParams,
+        investigationFields: migrateInvestigationFields(ruleParams.investigationFields),
+      };
       await validateBulkEditRule({
         mlAuthz,
-        ruleType: ruleParams.type,
+        ruleType: migratedRuleParams.type,
```

Comment:
I'm unsure about this change. Without it, doing a bulk action (edit) on a rule with the legacy investigation field fails on params validation. However, the registred params schema defined [here](https://github.com/elastic/kibana/pull/169061/files#diff-c6c196ee2805bef284eea67bc43f94a7fb9f3eeaa0dee2609df3617d2ccbf838R102) includes both the old and new versions, so I'm not sure why this change would be needed.

If someone could confirm my logic here before I merge, would be much appreciated. 

cc @banderror @vitaliidm @marshallmain @spong (pinging you Garrett since you were assigned to this PR).

Reply:
The error I get without this code here is `params invalid: ...` - complains of an array being passed for investigation fields. I traced the error to the bulkEdit rules client logic [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/application/rule/methods/bulk_edit/bulk_edit_rules.ts#L526). You can see that it validates the params against `ruleType.validate.params` - `ruleType` being the options passed when registering a rule type.

When we register the query rule type [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/plugin.ts#L290) - we use this [schema](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_schema/model/rule_schemas.ts#L92) in the [validation logic](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/query/create_query_alert_type.ts#L41).

This same schema used for validation is what is updated in this PR to allow for the investigation field to be both the old and new types.

Reply:
As far as I see the problem is in [this line](https://github.com/elastic/kibana/pull/169061/files#diff-afcaec4a7447adf94399594770f629a933be7d885a2828dfc1aa10b1acf6393fR298). NonEmptyStringArray expects a non empty string of values separated by commas. If you replace this line with
export const LegacyInvestigationFields = NonEmptyArray(NonEmptyString);
you can roll back the changes in x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/bulk_actions/bulk_edit_rules.ts and bulk actions will work.

Reply:
I was able to remove this logic after addressing the issue @maximpn spotted in the type definition.